src/Compilers/CSharp/Portable/BoundTree/Formatting.cs

```diff
@@ -148,9 +150,7 @@ public override object Display
     internal partial class BoundStackAllocArrayCreation
     {
         public override object Display
-        {
-            get { return string.Format(MessageID.IDS_StackAllocExpression.Localize().ToString(), ElementType, Count.Syntax); }
-        }
+            => ErrorFacts.GetMessageFormat(MessageID.IDS_StackAllocExpression, ElementType.ToDisplayString(), Count.Syntax.ToString());
```

Comment:
>ToDisplayString [](start = 91, length = 15)

That builds string too early. Nothing should be formatted until client asks for the string

Reply:
So we need a mixture of FormattableStringFactory.Create (delays the arguments) and ErrorFacts.GetMessageFormat (Delays the formatstring). I haven't seen anything that is able to do this. Can you give me a hint were to look?